app/jobs/check_fixity_recursive_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+class CheckFixityRecursiveJob < ApplicationJob
+  delegate :query_service, to: :metadata_adapter
+
+  def perform
+    file_set = query_service.custom_queries.least_recently_updated_file_set
+    original_file_metadata = file_set.run_fixity
+    file_set.file_metadata = file_set.file_metadata.select { |x| !x.original_file? } + Array.wrap(original_file_metadata)
+    metadata_adapter.persister.save(resource: file_set)
+    CheckFixityRecursiveJob.set(queue: queue_name).perform_later
```

Comment:
Should we catch everything above this line, so that if the job fails it's not over forever?

Reply:
good idea

Reply:
it will probably just keep trying again on the same file set? is that bad?

Reply:
Um. I'm not sure? It also strikes me that if it fails Sidekiq will try again - maybe it's better to just leave it that way.

Reply:
yeah; rubocop no likey and sidekiq should take care of it. i'll revert that change.